src/resources/uart_esp32_hal.c

```diff
@@ -112,6 +112,14 @@ void uart_toit_hal_set_mode(uart_hal_handle_t hal, uart_mode_t mode) {
   uart_hal_set_mode(HAL, mode);
 }
 
+bool IRAM_ATTR uart_toit_hal_is_tx_idle(uart_hal_handle_t hal) {
+  return uart_hal_is_tx_idle(HAL);
+}
+
+void IRAM_ATTR uart_toit_hal_set_rts(uart_hal_handle_t hal, bool active) {
+  uart_hal_set_rts(HAL, active ? 0 : 1);
```

Comment:
This surprised me a bit, but I deduced the `active` is zero from another UART driver.

Reply:
Interesting. Until now I didn't even realize that we would "bind" the RTS to the UART. I thought we would just set the pin directly.
No need for comments, but do you know if there is an advantage in using the `uart_hal` for the RTS, compared to just bit-banding the RTS by hand?